### PR TITLE
Dato Distributed job for parallel crawling of Hacker News articles

### PR DESCRIPTION
**Pull request by rlvoyer**

### CODE REVIEW

File: parallel_crawling_jobs.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+# Set this to a bucket to which you have write access
+s3_base_path = "s3://my-bucket/hacker_news/source_html"
+
+# Define a function to be called for each ID, URL pair
+@gl.deploy.required_packages(['requests == 2.3.0'])
+def get_source(id, url, s3_base_path, timeout=10):
+    import requests
+    import logging
+    import codecs
+    from graphlab.deploy._predictive_service._file_util import upload_to_s3
```
Comment (srikris) [inline parallel_crawling_jobs.py]: Don't use _ functions here here. Use boto or something like that to upload files to S3

File: parallel_crawling_jobs.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+# Get a list of ID, URL pairs from SFrame
+id_url_pairs = [(x["id"], x["url"]) for x in stories_sf if x["url"]]
+
+# The base path to where source articles are to be stored in S3
+# Set this to a bucket to which you have write access
+s3_base_path = "s3://my-bucket/hacker_news/source_html"
+
+# Define a function to be called for each ID, URL pair
+@gl.deploy.required_packages(['requests == 2.3.0'])
```
Comment (srikris) [inline parallel_crawling_jobs.py]: Add some comments to the function with an example

File: parallel_crawling_jobs.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            f.write(response_txt)
+        s3_path = s3_base_path + "/" + filename
+        upload_to_s3(filename, s3_path)
+        return s3_path
+    except requests.exceptions.Timeout:
+        logging.getLogger().warning("Request for (%s, %d) timed out" % (url, id))
+    except Exception as e:
+        logging.getLogger().warning("Unexpected error on article (%s, %d): %s" % (url, id, e.message))
+
+# Divvy up the list of ID, URL pairs to distribute among 4 workers
```
Comment (srikris) [inline parallel_crawling_jobs.py]: Amongst N workers.

File: parallel_crawling_jobs.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
+# Launch parallel jobs in EC2 for crawling web pages
+import graphlab as gl
```
Comment (srikris) [inline parallel_crawling_jobs.py]: Can you write a simple comment above here about how this file is organized.

File: parallel_crawling_jobs.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+def get_all_source(id_url_pairs):
+    results = []
+    for _id, url in id_url_pairs:
+        results.append(get_source(_id, url, s3_base_path))
+    return results
+
+# Specify EC2 execution environment
+# The 2nd parameter should be set to an S3 bucket to which you will write logs
+ec2 = gl.deploy.environment.EC2("ec2", "s3://my-bucket/logs")
+
```
Comment (srikris) [inline parallel_crawling_jobs.py]: You need to update README.md to add this to the list that gets displayed on the website.

File: parallel_crawling_jobs.py (hunk 3 above)
Comment (rlvoyer) [inline parallel_crawling_jobs.py]: Did you actually want me to change "among" to "amongst"? "amongst" is less common in American English.

File: parallel_crawling_jobs.py (hunk 1 above)
Comment (rlvoyer) [inline parallel_crawling_jobs.py]: Done.

File: parallel_crawling_jobs.py (hunk 5 above)
Comment (rlvoyer) [inline parallel_crawling_jobs.py]: Done.

**Pull request merged**